src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -195,7 +200,39 @@ internal string FullPath
             {
                 // Should already be canonicalized to a full path
                 ErrorUtilities.VerifyThrowInternalRooted(value);
-                _solutionFile = value;
+                if (FileUtilities.IsSolutionFilterFilename(value))
+                {
+                    using (JsonDocument text = JsonDocument.Parse(File.ReadAllText(value)))
+                    {
+                        try
+                        {
+                            JsonElement solution = text.RootElement.GetProperty("solution");
+                            _solutionFile = solution.GetProperty("path").GetString();
+                            _solutionFilter = new HashSet<string>(NativeMethodsShared.IsLinux ? StringComparer.Ordinal : StringComparer.OrdinalIgnoreCase);
```

Comment:
Where did we land on this case insensitivity/how does VS currently behave?

Reply:
If I modify a .sln file by changing case, VS handles it fine. If I modify a .slnf file to change case, it has trouble finding it. Since .sln files have existed much longer than .slnf files, I wouldn't be surprised if .slnf files change at some point, so I would prefer to stick with ignoring case.